src/components/PathwaysList/PathwaysList.tsx

```diff
@@ -36,19 +37,43 @@ const PathwaysList: FC<PathwaysListProps> = ({
   service,
   resources
 }) => {
+  const [criteria, setCriteria] = useState<CriteriaResult[] | null>(null);
+
+  if (!criteria && evaluatedPathways.length > 0 && resources != null && resources.length > 0) {
+    // Create a fake Bundle for the CQL engine and check if patientPath needs to be evaluated
+    const patient = {
+      resourceType: 'Bundle',
+      entry: resources.map((r: fhir.Resource) => ({ resource: r }))
+    };
+
+    // Evaluate pathway criteria for each pathway
+    const criteriaPromises: Promise<CriteriaResult>[] = [];
+    evaluatedPathways.forEach(pathway => {
+      criteriaPromises.push(evaluatePathwayCriteria(patient, pathway.pathway));
+    });
```

Comment:
Also not a big deal but this kind of thing is a good use for the `map` function:
```suggestion
    const criteriaPromises = evaluatedPathways.map(pathway => evaluatePathwayCriteria(patient, pathway.pathway));
```